test-utils/runtime/src/lib.rs

```diff
@@ -101,15 +101,36 @@ impl Transfer {
 	pub fn into_signed_tx(self) -> Extrinsic {
 		let signature = sp_keyring::AccountKeyring::from_public(&self.from)
 			.expect("Creates keyring from public key.").sign(&self.encode()).into();
-		Extrinsic::Transfer(self, signature)
+		Extrinsic::Transfer {
+			transfer: self,
+			signature,
+			exhaust_resources: false,
+		}
+	}
+
+	/// Convert into a signed extrinsic, which will not end up included
+	/// in block due to resource exhaustion.
+	#[cfg(feature = "std")]
+	pub fn into_exhaust_tx(self) -> Extrinsic {
+		let signature = sp_keyring::AccountKeyring::from_public(&self.from)
+			.expect("Creates keyring from public key.").sign(&self.encode()).into();
+		Extrinsic::Transfer {
+			transfer: self,
+			signature,
+			exhaust_resources: true,
+		}
 	}
 }
 
 /// Extrinsic for test-runtime.
 #[derive(Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
 pub enum Extrinsic {
 	AuthoritiesChange(Vec<AuthorityId>),
-	Transfer(Transfer, AccountSignature),
+	Transfer {
+		transfer: Transfer,
+		signature: AccountSignature,
+		exhaust_resources: bool,
```

Comment:
Why isn't this just a distinct variant in this enum?

Reply:
It has to share most of the logic with `Transfer`, since it has to come from some sender and have some nonce. I could simplify it to:
```
Extrinsic::ExhaustResources(from, nonce)
```
but I felt it's too arbitrary and it made more sense to just extend transfer.